src/ImageSharp/Processing/Processors/Transforms/ResizeProcessor.cs

```diff
@@ -40,25 +40,14 @@ public ResizeProcessor(ResizeOptions options, Size sourceSize)
             Guard.NotNull(options, nameof(options));
             Guard.NotNull(options.Sampler, nameof(options.Sampler));
 
-            int tempWidth = options.Size.Width;
-            int tempHeight = options.Size.Height;
+            int targetWidth = options.Size.Width;
+            int targetHeight = options.Size.Height;
 
             // Ensure size is populated across both dimensions.
             // These dimensions are used to calculate the final dimensions determined by the mode algorithm.
-            if (tempWidth == 0 && tempHeight > 0)
-            {
-                tempWidth = (int)MathF.Round(sourceSize.Width * tempHeight / (float)sourceSize.Height);
-            }
-
-            if (tempHeight == 0 && tempWidth > 0)
-            {
-                tempHeight = (int)MathF.Round(sourceSize.Height * tempWidth / (float)sourceSize.Width);
-            }
-
-            Guard.MustBeGreaterThan(tempWidth, 0, nameof(tempWidth));
-            Guard.MustBeGreaterThan(tempHeight, 0, nameof(tempHeight));
+            EnsureSizeBothDimensions(sourceSize.Width, sourceSize.Height, ref targetWidth, ref targetHeight, out _, out _);
```

Comment:
I added EnsureSizeBothDimensions to deduplicate code in two constructors. I can remove the method and reinstate the duplicated blocks if you prefer.

But about the Guards, I think they are still needed in case <i>both</i> targetwidth and targetheight is specified to be zero, right? My changes is only relevant when only one target dimension is zero.

Reply:
Sorry, yeah keep the Guards. I think duplication is ok in this instance. The methods signature for `EnsureSizeBothDimensions` just doesn't feel right to me.

Reply:
Yes that method is a bit clumsy. I've done the changes.